src/github.com/matrix-org/dendrite/roomserver/storage/sql.go

```diff
@@ -230,6 +237,11 @@ const insertEventStateKeyNIDSQL = "" +
 const selectEventStateKeyNIDSQL = "" +
 	"SELECT event_state_key_nid FROM event_state_keys WHERE event_state_key = $1"
 
+const selectEventStateKeyNIDsSQL = "" +
+	"SELECT event_state_key, event_state_key_nid FROM event_state_keys" +
+	" WHERE event_state_key = ANY($1)" +
```

Comment:
It is similar but the parameter is an array. https://www.postgresql.org/docs/9.1/static/arrays.html
This means that the entire list is passed as a parameter to the query rather than being part of the query string. I've yet to find a limit on the size of query parameter in postgresql.